packages/core/src/components/editable-text/editableText.tsx

```diff
@@ -212,27 +212,19 @@ export class EditableText extends AbstractComponent<IEditableTextProps, IEditabl
     }
 
     public cancelEditing = () => {
-        const { lastValue, value } = this.state;
-        if (lastValue === value) {
-            this.setState({ isEditing: false });
-        } else {
-            this.setState({ isEditing: false, value: lastValue });
-            // invoke onCancel after onChange so consumers' onCancel can override their onChange
-            safeInvoke(this.props.onChange, lastValue);
-            safeInvoke(this.props.onCancel, lastValue);
-        }
+        const { lastValue } = this.state;
+        this.setState({ isEditing: false, value: lastValue });
+        // invoke onCancel after onChange so consumers' onCancel can override their onChange
+        safeInvoke(this.props.onChange, lastValue);
```

Comment:
Does it make more sense to at least not trigger the `this.props.onChange` callback unless `if (lastValue === value)`?